### PR TITLE
Only push reset-before-request to version GE 1.23

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/networking/core/route/route.go [hunk 1]
```diff
@@ -643,6 +644,18 @@ func applyHTTPRouteDestination(
 		}
 	}
 	action.RetryPolicy = retry.ConvertPolicy(policy, consistentHash)
+	// TODO: Remove two or more releases later
+	if strings.Contains(action.RetryPolicy.RetryOn, "reset-before-request") {
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/route/route.go:648]: I do not think this is needed. Envoy will just ignore this even if we send to older proxies

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/route/route.go:648]: If that true, it would be quite good.

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/route/route.go:648]: Not sure why the `checkHeades` will validate the retryON
```
  if (config_->strict_check_headers_ != nullptr) {
    for (const auto& header : *config_->strict_check_headers_) {
      const auto res = FilterUtility::StrictHeaderChecker::checkHeader(headers, header);
      if (!res.valid_) {
        callbacks_->streamInfo().setResponseFlag(
            StreamInfo::CoreResponseFlag::InvalidEnvoyRequestHeaders);
        const std::string body = fmt::format("invalid header '{}' with value '{}'",
                                             std::string(res.entry_->key().getStringView()),
                                             std::string(res.entry_->value().getStringView()));
        const std::string details =
            absl::StrCat(StreamInfo::ResponseCodeDetails::get().InvalidEnvoyRequestHeaders, "{",
                         StringUtil::replaceAllEmptySpace(res.entry_->key().getStringView()), "}");
        callbacks_->sendLocalReply(Http::Code::BadRequest, body, nullptr, absl::nullopt, details);
        return Http::FilterHeadersStatus::StopIteration;
      }
    }
  }
```

Comment (ramaraochavali) [reply pilot/pkg/networking/core/route/route.go:648]: Envoy will only validate if it is passed via x-envoy-retry-on header with strict header matching on.

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/route/route.go:648]: got it

**Pull request closed without merging**